app/src/main/java/com/github/se/eduverse/ui/todo/TodoListScreen.kt

```diff
@@ -0,0 +1,164 @@
+package com.github.se.eduverse.ui.todo
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material.icons.rounded.Done
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.se.eduverse.model.Todo
+import com.github.se.eduverse.model.TodoStatus
+import com.github.se.eduverse.ui.navigation.NavigationActions
+import com.github.se.eduverse.ui.navigation.TopNavigationBar
+import com.github.se.eduverse.viewmodel.TodoListViewModel
+
+/** Composable that represents the user's todo list screen */
+@Composable
+fun TodoListScreen(
+    navigationActions: NavigationActions,
+    todoListViewModel: TodoListViewModel = viewModel(factory = TodoListViewModel.Factory)
```

Comment:
I think it would be better to defined a todoListViewModel in EduverseApp and use it there. This way it won't build a new instance everytime we open this screen, and it gives better consistency regarding to the way the other screens are called

Reply:
Ok, I'll change it